### ISSUE TITLE
Object-centric Learning with Cyclic Walks between Parts and Whole

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#227]
# TL; DR

- Learn part-whole correspondence in images in a self-supervised manner.

# Motivation

- Object-centric representation learning

# Formulation

![image](https://github.com/reyllama/paper-reviews/assets/46308407/d950e34a-b2c0-4ce9-8455-74323e232bff)

### Slot-Attention

- Cross-attention layer that forms an attentional information bottleneck (similar to attention poolers)

### Contrastive Random Walk

- In video, learn low-level correspondence between image regions of adjacent frames.
- In image, learn part-whole correspondence where ***whole*** features are extracted through the slot-attention (object-centric).

# Evaluation

- Though the idea is very compelling, evaluation is a little weak in my opinion.
- Performance tuning with these frameworks still requires much effort.
- Model components are ablated neatly.
- Temperature **matters** in contrastive learning (InfoNCE form).

![image](https://github.com/reyllama/paper-reviews/assets/46308407/9c1b3ba4-46fc-430f-b780-3b943d9313d2)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/a7e91712-c15f-4b68-abc2-ec3b7c883196)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/b60beb4e-62a1-4624-9957-659b43d568d4)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/54a36de4-4908-4934-bdac-5d82c6708785)

# Inspiration

- Should go through h.p. sweep on the temperature, in the least.
- Slot-attention is a rather elegant way of aggregating image features (into a predefined level of granularity, set by the number of queries in the CA layer).
